tests/lib/uuid/src/main.c

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) 2024, SECO Mind Srl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/ztest.h>
+#include <zephyr/sys/uuid.h>
+
+#ifdef CONFIG_UUID_V4
+ZTEST(uuid, test_uuid_v4)
+{
+	struct uuid gen_uuid = {0};
+
+	int result = uuid_generate_v4(&gen_uuid);
+	/* Check version and variant fields */
+	zassert_true(result == 0, "uuid_generate_v4 returned an error");
+	zassert_equal(gen_uuid.val[6U] >> 4U, 4U,
+		      "Generated UUID v4 contains an incorrect 'ver' field");
+	zassert_equal(gen_uuid.val[8U] >> 6U, 2U,
+		      "Generated UUID v4 contains an incorrect 'var' field");
+}
+#else
+ZTEST(uuid, test_uuid_v4)
+{
+	ztest_test_skip();
+}
+#endif
```

Comment:
```suggestion
#endif
```

It doesn't really make sense to have the `#else` here, does it? 

You should also be able to just add something like `Z_TEST_SKIP_IFNDEF(CONFIG_UUID_V4)` in the test itself